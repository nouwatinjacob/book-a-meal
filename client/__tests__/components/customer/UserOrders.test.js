import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  UserOrder,
  mapStateToProps
} from '../../../src/components/customer/UserOrders.jsx';
import {
  userOrderResponse
} from '../../__mockData__/mockOrder';

const props = {
  getUserOrderAction: jest.fn(),
  cancelOrderAction: () => Promise.resolve(),
  isSuccessful: true,
  orderState: {
    orders: {
      userOrderResponse,
      paginate: userOrderResponse.paginate
    }
  }
};

const ids = [1, 2, 3];

describe('UserOrder component', () => {
  localStorage.setItem('token', customerToken);
  sessionStorage.setItem('ids', ids);
  const setup = () => shallow(<UserOrder {...props} />);

  it('should render ModifyOrder component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'handleModify'
  if modify order button if clicked`, (done) => {
    const wrapper = setup();

    const mealId = 1;
    const menuId = 1;
    const orderId = 1;

    const handleModifySpy = jest.spyOn(
      wrapper.instance(),
      'handleModify'
    );
    wrapper.instance().handleModify(mealId, menuId, orderId);
    expect(handleModifySpy).toHaveBeenCalled();
    done();
  });

  it(`should call 'handleCancelOrder'
  when modify order button is clicked`, (done) => {
    const wrapper = setup();

    const orderId = 1;

    const handleCancelOrderSpy = jest.spyOn(
      wrapper.instance(),
      'handleCancelOrder'
    );
    wrapper.instance().handleCancelOrder(orderId);
    expect(handleCancelOrderSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'handlePageClick'
  if pagination button is clicked`, (done) => {
    const wrapper = setup();

    const data = {
      selected: 1
    };

    const handlePageClickSpy = jest.spyOn(
      wrapper.instance(),
      'handlePageClick'
    );
    wrapper.instance().handlePageClick(data);
    expect(handlePageClickSpy).toHaveBeenCalled();

    done();
  });
});