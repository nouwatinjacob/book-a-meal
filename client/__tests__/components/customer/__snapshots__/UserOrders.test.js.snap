// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserOrder component should render ModifyOrder component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <ToastContainer
    autoClose={5000}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnHover={true}
    pauseOnVisibilityChange={true}
    position="top-right"
    progressClassName={null}
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <div
    className="wrapper"
  >
    <h1
      className="text-center"
    >
      Order List
    </h1>
    <h3
      className="text-center"
    >
      You have no Order
    </h3>
  </div>
  <br />
  <br />
</div>
`;
